module/mailnews/newsread.c

```diff
@@ -383,14 +399,74 @@ Bool DateFromSeconds(long seconds, char *str)
    struct tm *t;
    time_t local_time = seconds;
 
-   local_time += 1534000000L;    // Offset to sometime in mid-2018
+   local_time += 1534000000L; // Offset to sometime in mid-2018
    t = localtime(&local_time);
 
    if (t == NULL)
       return False;
 
-   sprintf(str, "%s %s %.2ld, %.4ld %.2ld:%.2ld", 
-	   GetString(hInst, weekdays[t->tm_wday]), GetString(hInst, months[t->tm_mon]), 
-	   t->tm_mday, t->tm_year + 1900, t->tm_hour, t->tm_min);
+   sprintf(str, "%s %s %.2ld, %.4ld %.2ld:%.2ld",
+           GetString(hInst, weekdays[t->tm_wday]), GetString(hInst, months[t->tm_mon]),
+           t->tm_mday, t->tm_year + 1900, t->tm_hour, t->tm_min);
    return True;
 }
+
+void OnColumnClick(LPNMLISTVIEW pLVInfo)
+{
+   static int nSortColumn = 0;
+   static BOOL bSortAscending = TRUE;
+   LPARAM lParamSort;
+
+   // get new sort parameters
+   if (pLVInfo->iSubItem == nSortColumn)
+      bSortAscending = !bSortAscending;
+   else
+   {
+      nSortColumn = pLVInfo->iSubItem;
+      bSortAscending = TRUE;
+   }
+
+   // combine sort info into a single value we can send to our sort function
+   lParamSort = 1 + nSortColumn;
+   if (!bSortAscending)
+      lParamSort = -lParamSort;
+
+   // sort list
+   ListView_SortItems(pLVInfo->hdr.hwndFrom, CompareListItems, lParamSort);
+}
+
+int CALLBACK CompareListItems(LPARAM lParam1, LPARAM lParam2, LPARAM lParamSort)
+{
+   NewsArticle *article1, *article2;
+   BOOL bSortAscending = (lParamSort > 0);
+   int nColumn = abs(lParamSort) - 1;
+
+   article1 = (NewsArticle *) lParam1;
```

Comment:
We use the C++ compiler, so you can declare variables at first use, which is better practice.  i.e.

```
NewsArticle *article1 = (NewsArticle *) lParam1;
```